android/src/main/java/com/applovin/reactnative/AppLovinMAXModule.java

```diff
@@ -1045,6 +1089,41 @@ public void onAdCollapsed(final MaxAd ad)
         sendReactNativeEvent( ( MaxAdFormat.MREC == adFormat ) ? "OnMRecAdCollapsedEvent" : "OnBannerAdCollapsedEvent", getAdInfo( ad ) );
     }
 
+    @Override
+    public void onAdRevenuePaid(final MaxAd ad)
+    {
+        final MaxAdFormat adFormat = ad.getFormat();
+        final String name;
+        if ( MaxAdFormat.BANNER == adFormat || MaxAdFormat.LEADER == adFormat )
+        {
+            name = "OnBannerAdRevenuePaid";
+        }
+        else if ( MaxAdFormat.MREC == adFormat )
+        {
+            name = "OnMRecAdRevenuePaid";
+        }
+        else if ( MaxAdFormat.INTERSTITIAL == adFormat )
+        {
+            name = "OnInterstitialAdRevenuePaid";
+        }
+        else if ( MaxAdFormat.REWARDED == adFormat )
+        {
+            name = "OnRewardedAdRevenuePaid";
+        }
+        else
+        {
+            logInvalidAdFormat( adFormat );
+            return;
+        }
+
+        WritableMap adInfo = getAdInfo( ad );
+        adInfo.putString( "networkPlacement", !TextUtils.isEmpty( ad.getNetworkPlacement() ) ? ad.getNetworkPlacement() : "" );
+        adInfo.putString( "revenuePrecision", !TextUtils.isEmpty( ad.getRevenuePrecision() ) ? ad.getRevenuePrecision() : "" );
+        adInfo.putString( "countryCode", !TextUtils.isEmpty( sdkConfiguration.getCountryCode() ) ? sdkConfiguration.getCountryCode() : "" );
```

Comment:
Will never be null, so no need for the extra code to check.

Reply:
Fixed iOS as well.